### PR TITLE
Properly set the name for C++ overload set instructions in SemIR

### PR DESCRIPTION
**Pull request by bricknerb**
This is a followup of #5891.
Part of #5915.

### CODE REVIEW

File: toolchain/check/testdata/interop/cpp/function/multiple_too_few_args_calls.carbon [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/none.carbon
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interop/cpp/function/multiple_too_few_args_calls.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/interop/cpp/function/multiple_too_few_args_calls.carbon
+
+// --- fail_call_too_few_args.carbon
```
Comment (ivanaivanovska) [inline toolchain/check/testdata/interop/cpp/function/multiple_too_few_args_calls.carbon:13]: Consider moving this test to an existing file e.g. `overloads.carbon`, there are already several tests for multiple calls, adding this one could fit well there.

**Review by ivanaivanovska — APPROVED** [review]

File: toolchain/check/testdata/interop/cpp/function/multiple_too_few_args_calls.carbon (hunk 1 above)
Comment (bricknerb) [reply toolchain/check/testdata/interop/cpp/function/multiple_too_few_args_calls.carbon:13]: The tests for multiple calls in `overloads.carbon` are `import_both_overloaded_functions_called` `import_multiple_overloaded_sets`, both of them focus on functions with overloads.
This test functionality doesn't really focus on overloading.
It focuses on calling a function with wrong number of args, which is tested in many test files: `cpp_diagnostics`, `decayed_param`, `default_arg`, `method`, `operators`, `overloads` and `reference`.
Calling more than one function in a test is covered by many tests, so I don't think this makes `overloads` a great fit, and I think a reader would find it strange to have a test with no overloads there.
I think this use case is somewhat special since it catches a crash that happens on a combination of circumstances, so I definitely want to have it to make sure the crash doesn't recur and putting it separate as it's not a natural fit elsewhere might help it outlive a cleanup.

**Review by geoffromer — APPROVED** [review]
